src/cmakeProject.ts

```diff
@@ -987,9 +987,14 @@ export class CMakeProject {
 
     async initializeKitOrPresets() {
         if (this.useCMakePresets) {
-            const configurePreset = this.workspaceContext.state.configurePresetName;
-            if (configurePreset) {
-                await this.presetsController.setConfigurePreset(configurePreset);
+            const latestConfigPresetName = this.workspaceContext.state.configurePresetName;
+            if (latestConfigPresetName) {
+                // Check if the latest configurePresetName from the previous session is still valid.
+                const presets = await this.presetsController.getAllConfigPresets();
+                const latestConfigPreset: preset.ConfigurePreset | undefined = presets.find(preset => preset.name === latestConfigPresetName);
+                if (latestConfigPreset && !latestConfigPreset.hidden) {
```

Comment:
It seems like if the latestConfigPresetName isn't valid anymore, we are left without a preset being set. Should we have a fall back behavior for the case where latestConfigPreset is 'undefined'?

Reply:
This is the initialization step when we load the project, so if we don't initialize it, it will stay undefined and the user will have to pick a new preset.

Reply:
Okay. So the user will be prompted to set a new preset rather than have the previous preset set as being enabled?

Reply:
correct.